In [1]:
import pandas as pd
from matplotlib import pyplot as plt
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
import seaborn as sns

In [2]:
datos = pd.read_csv('historical_data.csv')

In [3]:
datos

market_id           created_at actual_delivery_time  store_id  \
0             1.0  2015-02-06 22:24:17  2015-02-06 23:27:16      1845   
1             2.0  2015-02-10 21:49:25  2015-02-10 22:56:29      5477   
2             3.0  2015-01-22 20:39:28  2015-01-22 21:09:09      5477   
3             3.0  2015-02-03 21:21:45  2015-02-03 22:13:00      5477   
4             3.0  2015-02-15 02:40:36  2015-02-15 03:20:26      5477   
...           ...                  ...                  ...       ...   
197423        1.0  2015-02-17 00:19:41  2015-02-17 01:24:48      2956   
197424        1.0  2015-02-13 00:01:59  2015-02-13 00:58:22      2956   
197425        1.0  2015-01-24 04:46:08  2015-01-24 05:36:16      2956   
197426        1.0  2015-02-01 18:18:15  2015-02-01 19:23:22      3630   
197427        1.0  2015-02-08 19:24:33  2015-02-08 20:01:41      3630   

       store_primary_category  order_protocol  total_items  subtotal  \
0                    american             1.0            4      3441   
1                     mexican             2.0            1      1900   
2                         NaN             1.0            1      1900   
3                         NaN             1.0            6      6900   
4                         NaN             1.0            3      3900   
...                       ...             ...          ...       ...   
197423                   fast             4.0            3      1389   
197424                   fast             4.0            6      3010   
197425                   fast             4.0            5      1836   
197426               sandwich             1.0            1      1175   
197427               sandwich             1.0            4      2605   

        num_distinct_items  min_item_price  max_item_price  \
0                        4             557            1239   
1                        1            1400            1400   
2                        1            1900            1900   
3                        5             600            1800   
4                        3            1100            1600   
...                    ...             ...             ...   
197423                   3             345             649   
197424                   4             405             825   
197425                   3             300             399   
197426                   1             535             535   
197427                   4             425             750   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
0                        33.0                14.0                      21.0   
1                         1.0                 2.0                       2.0   
2                         1.0                 0.0                       0.0   
3                         1.0                 1.0                       2.0   
4                         6.0                 6.0                       9.0   
...                       ...                 ...                       ...   
197423                   17.0                17.0                      23.0   
197424                   12.0                11.0                      14.0   
197425                   39.0                41.0                      40.0   
197426                    7.0                 7.0                      12.0   
197427                   20.0                20.0                      23.0   

        estimated_order_place_duration  \
0                                  446   
1                                  446   
2                                  446   
3                                  446   
4                                  446   
...                                ...   
197423                             251   
197424                             251   
197425                             251   
197426                             446   
197427                             446   

        estimated_store_to_consumer_driving_duration  
0           

pregunta 2

In [4]:
datos_pizza=datos[datos['store_primary_category']=="pizza"]

filtramos para que solo aparescan registro de pizza

In [5]:
datos_pizza

market_id           created_at actual_delivery_time  store_id  \
74            2.0  2015-02-13 03:48:58  2015-02-13 04:26:51      2004   
249           4.0  2015-01-25 21:27:26  2015-01-25 22:10:14      5081   
897           3.0  2015-01-25 02:32:10  2015-01-25 03:32:24      6922   
898           3.0  2015-02-18 03:22:51  2015-02-18 03:49:18      6922   
899           3.0  2015-01-23 02:05:52  2015-01-23 02:48:35      6922   
...           ...                  ...                  ...       ...   
196897        1.0  2015-02-07 03:11:02  2015-02-07 03:44:59      1520   
197126        2.0  2015-01-23 02:52:55  2015-01-23 03:36:09      1409   
197222        3.0  2015-01-28 05:13:06  2015-01-28 05:38:48      1340   
197276        5.0  2015-01-23 04:27:25  2015-01-23 05:09:22      1340   
197323        4.0  2015-02-13 20:46:24  2015-02-13 21:34:13      6110   

       store_primary_category  order_protocol  total_items  subtotal  \
74                      pizza             2.0            4      2300   
249                     pizza             5.0            1       950   
897                     pizza             1.0            5      6900   
898                     pizza             1.0            2      1950   
899                     pizza             1.0            2      3600   
...                       ...             ...          ...       ...   
196897                  pizza             3.0            3      3785   
197126                  pizza             4.0            3      1965   
197222                  pizza             4.0            1       699   
197276                  pizza             4.0            3      2057   
197323                  pizza             2.0            2      2300   

        num_distinct_items  min_item_price  max_item_price  \
74                       3             400             900   
249                      1             950             950   
897                      5             900            1800   
898                      2             350            1600   
899                      2            1800            1800   
...                    ...             ...             ...   
196897                   3             395            1895   
197126                   2             375             795   
197222                   1             699             699   
197276                   3             659             699   
197323                   2            1025            1275   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
74                       32.0                30.0                      37.0   
249                      65.0                71.0                      29.0   
897                       8.0                 7.0                       6.0   
898                       9.0                 7.0                       6.0   
899                       7.0                 7.0                       7.0   
...                       ...                 ...                       ...   
196897                  132.0               109.0                     182.0   
197126                  111.0                71.0                     115.0   
197222                   17.0                13.0                      13.0   
197276                   36.0                32.0                      37.0   
197323                   64.0                64.0                      85.0   

        estimated_order_place_duration  \
74                                 251   
249                                251   
897                                446   
898                                446   
899                                446   
...                                ...   
196897                             251   
197126                             251   
197222                             251   
197276                             251   
197323                             251   

        estimated_store_to_consumer_driving_duration  
74          

In [6]:
datos2=datos_pizza[['store_id' ]]

tomamos todas als tiendas que venden pizzas

In [7]:
datos2

store_id
74          2004
249         5081
897         6922
898         6922
899         6922
...          ...
196897      1520
197126      1409
197222      1340
197276      1340
197323      6110

[17321 rows x 1 columns]

In [8]:
datos2=datos2.drop_duplicates()

In [9]:
datos2

store_id
74          2004
249         5081
897         6922
1129        4616
1274        6313
...          ...
196800      6964
196879      1520
197126      1409
197222      1340
197323      6110

[1088 rows x 1 columns]

al solo tener una colummna eliminas todos las filas repetidas y asi tenemos el cuantas tiendas vende pizzas

In [10]:
datos2.count()

store_id    1088
dtype: int64

contamos cuantas tiendas diferentes venden pizzas en total

pregunta 3

In [11]:
datos3=datos[['num_distinct_items','estimated_store_to_consumer_driving_duration']]

tomamos las colummnas que vamos a analizar

In [12]:
datos3

num_distinct_items  estimated_store_to_consumer_driving_duration
0                        4                                         861.0
1                        1                                         690.0
2                        1                                         690.0
3                        5                                         289.0
4                        3                                         650.0
...                    ...                                           ...
197423                   3                                         331.0
197424                   4                                         915.0
197425                   3                                         795.0
197426                   1                                         384.0
197427                   4                                         134.0

[197428 rows x 2 columns]

In [13]:
datos3.corr()

num_distinct_items  \
num_distinct_items                                      1.000000   
estimated_store_to_consumer_driving_duration            0.022693   

                                              estimated_store_to_consumer_driving_duration  
num_distinct_items                                                                0.022693  
estimated_store_to_consumer_driving_duration                                      1.000000

como se puede observar el tiempo de viaje estimado entre la tienda y el consumidor no estan asociados al numero de articulos distintos en la compra, ya que podemosen la matriz de correlacion que la relacion es muy baja entre estas variuables, ya que presenta un valor de 0.022693 un valor muy cercano a  0, y lo ideal es lo mas cercano a 1 o -1.

si hacemos unas colsutas podemos ver que el registro 1 tiene  numero distinto de productos asi como 1 total de productos y tiene una duracion de 690 mientras que otro con un total de 6 productos y 5 son distintos y tiene una duracion de 289, o otro con 3 productos distintos y total 3 y tiene una duracion de 331.

pregunta 4

In [14]:
datos4=datos
datos4['delivery_time'] = (pd.to_datetime(datos['actual_delivery_time']) - pd.to_datetime(datos['created_at'])).dt.total_seconds()

creamos una nueva columna que sera el resultado de los tiempo entre que se pide el pedido y el tiempo que se entrega al cliente

In [15]:
datos4 = datos4.dropna(axis=0)
datos4

market_id           created_at actual_delivery_time  store_id  \
0             1.0  2015-02-06 22:24:17  2015-02-06 23:27:16      1845   
1             2.0  2015-02-10 21:49:25  2015-02-10 22:56:29      5477   
8             2.0  2015-02-16 00:11:35  2015-02-16 00:38:01      5477   
14            1.0  2015-02-12 03:36:46  2015-02-12 04:14:39      2841   
15            1.0  2015-01-27 02:12:36  2015-01-27 03:02:24      2841   
...           ...                  ...                  ...       ...   
197423        1.0  2015-02-17 00:19:41  2015-02-17 01:24:48      2956   
197424        1.0  2015-02-13 00:01:59  2015-02-13 00:58:22      2956   
197425        1.0  2015-01-24 04:46:08  2015-01-24 05:36:16      2956   
197426        1.0  2015-02-01 18:18:15  2015-02-01 19:23:22      3630   
197427        1.0  2015-02-08 19:24:33  2015-02-08 20:01:41      3630   

       store_primary_category  order_protocol  total_items  subtotal  \
0                    american             1.0            4      3441   
1                     mexican             2.0            1      1900   
8                      indian             3.0            4      4771   
14                    italian             1.0            1      1525   
15                    italian             1.0            2      3620   
...                       ...             ...          ...       ...   
197423                   fast             4.0            3      1389   
197424                   fast             4.0            6      3010   
197425                   fast             4.0            5      1836   
197426               sandwich             1.0            1      1175   
197427               sandwich             1.0            4      2605   

        num_distinct_items  min_item_price  max_item_price  \
0                        4             557            1239   
1                        1            1400            1400   
8                        3             820            1604   
14                       1            1525            1525   
15                       2            1425            2195   
...                    ...             ...             ...   
197423                   3             345             649   
197424                   4             405             825   
197425                   3             300             399   
197426                   1             535             535   
197427                   4             425             750   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
0                        33.0                14.0                      21.0   
1                         1.0                 2.0                       2.0   
8                         8.0                 6.0                      18.0   
14                        5.0                 6.0                       8.0   
15                        5.0                 5.0                       7.0   
...                       ...                 ...                       ...   
197423                   17.0                17.0                      23.0   
197424                   12.0                11.0                      14.0   
197425                   39.0                41.0                      40.0   
197426                    7.0                 7.0                      12.0   
197427                   20.0                20.0                      23.0   

        estimated_order_place_duration  \
0                                  446   
1                                  446   
8                                  446   
14                                 446   
15                                 446   
...                                ...   
197423                             251   
197424                             251   
197425                             251   
197426                             446   
197427                             446   

        estimated_store_to_consumer_driving_duration  delivery_time

eliminamos las fillas null

In [16]:
datosg1=datos4.groupby('store_primary_category')
datos4['prom']=datosg1['delivery_time'].transform('mean')
datos4.min()

<ipython-input-16-53172b657798>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  datos4['prom']=datosg1['delivery_time'].transform('mean')


market_id                                                       1.0
created_at                                      2015-01-21 15:22:03
actual_delivery_time                            2015-01-21 16:16:34
store_id                                                          1
store_primary_category                                       afghan
order_protocol                                                  1.0
total_items                                                       1
subtotal                                                          0
num_distinct_items                                                1
min_item_price                                                  -86
max_item_price                                                    0
total_onshift_dashers                                          -4.0
total_busy_dashers                                             -5.0
total_outstanding_orders                                       -6.0
estimated_order_place_duration                  

Agrupamos por diferentes tipos de comida para asi sacar el promedio de cuanto se tarda cada comida



buscamos el valor minimo de de los tiempos '2048'

In [17]:
datos5=datos4
datos6=datos5

In [18]:
datos6=datos5.set_index('prom', inplace=True)

ponemos los tiempos como indices

In [19]:
datos5.loc[2048]

market_id                                                       4.0
created_at                                      2015-01-24 00:42:24
actual_delivery_time                            2015-01-24 01:16:32
store_id                                                       6165
store_primary_category                                    chocolate
order_protocol                                                  1.0
total_items                                                       1
subtotal                                                       2650
num_distinct_items                                                1
min_item_price                                                 2650
max_item_price                                                 2650
total_onshift_dashers                                          52.0
total_busy_dashers                                             39.0
total_outstanding_orders                                       38.0
estimated_order_place_duration                  

dado el menos tiempo buscamos el registro con el menor tiempo posible, asi obtenemos el tipo de comida con el menor tiempo posible siendo chocolate

pregunta 1

In [20]:
datos6=pd.read_csv('historical_data.csv')
datos6=datos6.dropna(axis=0)
datos7=datos.groupby('store_id')
datos8=datos6

Hacemos un grupo para las tiendas  y limpiamos los datos eliminando los datos null

In [21]:
datos8['sum'] = datos7['subtotal'].transform('sum')

creamos una fila donde contamos el total de ventas de cada tienda

In [22]:
datos8

market_id           created_at actual_delivery_time  store_id  \
0             1.0  2015-02-06 22:24:17  2015-02-06 23:27:16      1845   
1             2.0  2015-02-10 21:49:25  2015-02-10 22:56:29      5477   
8             2.0  2015-02-16 00:11:35  2015-02-16 00:38:01      5477   
14            1.0  2015-02-12 03:36:46  2015-02-12 04:14:39      2841   
15            1.0  2015-01-27 02:12:36  2015-01-27 03:02:24      2841   
...           ...                  ...                  ...       ...   
197423        1.0  2015-02-17 00:19:41  2015-02-17 01:24:48      2956   
197424        1.0  2015-02-13 00:01:59  2015-02-13 00:58:22      2956   
197425        1.0  2015-01-24 04:46:08  2015-01-24 05:36:16      2956   
197426        1.0  2015-02-01 18:18:15  2015-02-01 19:23:22      3630   
197427        1.0  2015-02-08 19:24:33  2015-02-08 20:01:41      3630   

       store_primary_category  order_protocol  total_items  subtotal  \
0                    american             1.0            4      3441   
1                     mexican             2.0            1      1900   
8                      indian             3.0            4      4771   
14                    italian             1.0            1      1525   
15                    italian             1.0            2      3620   
...                       ...             ...          ...       ...   
197423                   fast             4.0            3      1389   
197424                   fast             4.0            6      3010   
197425                   fast             4.0            5      1836   
197426               sandwich             1.0            1      1175   
197427               sandwich             1.0            4      2605   

        num_distinct_items  min_item_price  max_item_price  \
0                        4             557            1239   
1                        1            1400            1400   
8                        3             820            1604   
14                       1            1525            1525   
15                       2            1425            2195   
...                    ...             ...             ...   
197423                   3             345             649   
197424                   4             405             825   
197425                   3             300             399   
197426                   1             535             535   
197427                   4             425             750   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
0                        33.0                14.0                      21.0   
1                         1.0                 2.0                       2.0   
8                         8.0                 6.0                      18.0   
14                        5.0                 6.0                       8.0   
15                        5.0                 5.0                       7.0   
...                       ...                 ...                       ...   
197423                   17.0                17.0                      23.0   
197424                   12.0                11.0                      14.0   
197425                   39.0                41.0                      40.0   
197426                    7.0                 7.0                      12.0   
197427                   20.0                20.0                      23.0   

        estimated_order_place_duration  \
0                                  446   
1                                  446   
8                                  446   
14                                 446   
15                                 446   
...                                ...   
197423                             251   
197424                             251   
197425                             251   
197426                             446   
197427                             446   

        estimated_store_to_consumer_driving_duration     sum  
0   

In [23]:
datos8.max()

market_id                                                       6.0
created_at                                      2015-02-18 06:00:44
actual_delivery_time                            2015-02-19 22:45:31
store_id                                                       6987
store_primary_category                                   vietnamese
order_protocol                                                  7.0
total_items                                                     411
subtotal                                                      26800
num_distinct_items                                               20
min_item_price                                                14700
max_item_price                                                14700
total_onshift_dashers                                         171.0
total_busy_dashers                                            154.0
total_outstanding_orders                                      285.0
estimated_order_place_duration                  

vemos el monto de mayor ventas de todas

In [24]:
datos9=datos8[['store_id','sum']]


separamos las tiendas y cuantos vendieron en total

In [25]:
datos9

store_id     sum
0           1845    3441
1           5477   48607
8           5477   48607
14          2841   17145
15          2841   17145
...          ...     ...
197423      2956  121765
197424      2956  121765
197425      2956  121765
197426      3630    3780
197427      3630    3780

[175777 rows x 2 columns]

In [26]:
datos10=datos9.set_index('sum', inplace=True)

In [27]:
datos9

store_id
sum             
3441        1845
48607       5477
48607       5477
17145       2841
17145       2841
...          ...
121765      2956
121765      2956
121765      2956
3780        3630
3780        3630

[175777 rows x 1 columns]

ponemos como indice el el total de ventas

In [28]:
datos9=datos9.drop_duplicates()
datos9.loc[2636241]

store_id    6074
Name: 2636241, dtype: int64

eliminamos los duplicados de las tiendas y asi solo tenemos una tienda con un total de ventas y depsues buscamos con la funcion loc la tienda con la mayor ventas de todas que fue la tienda id 6074

Pregunta 5

In [29]:
datos6=pd.read_csv('historical_data.csv')
datos6=datos6.dropna(axis=0)
datos11=datos.groupby('market_id')
datos12=datos6

agrupamos todas las tiendas

In [30]:
datos12['sum'] = datos11['market_id'].transform('count')

contamos cuantos pedidos tienen por region

In [31]:
datos12.min()

market_id                                                       1.0
created_at                                      2015-01-21 15:22:03
actual_delivery_time                            2015-01-21 16:16:34
store_id                                                          1
store_primary_category                                       afghan
order_protocol                                                  1.0
total_items                                                       1
subtotal                                                          0
num_distinct_items                                                1
min_item_price                                                  -86
max_item_price                                                    0
total_onshift_dashers                                          -4.0
total_busy_dashers                                             -5.0
total_outstanding_orders                                       -6.0
estimated_order_place_duration                  

vemos el valor minimo del resultado de la cantidad de pedidos en cada region

In [32]:
datos13=datos12[['market_id','sum']]

separamos los datos market_id y el contador

In [33]:
datos14=datos13.set_index('sum', inplace=True)
datos13=datos13.drop_duplicates()
datos13

market_id
sum               
38037.0        1.0
55058.0        2.0
23297.0        3.0
47599.0        4.0
18000.0        5.0
14450.0        6.0

ponemos como indices los contadores y despues eliminamos los duplicados

obtenemos la region con menos pedidos de todas siendo la region 6

In [34]:
datos12['times']= pd.to_datetime(datos12['created_at']).dt.hour

creamos una nueva columna que seran la hora del dia del pedido

In [35]:
datos12

market_id           created_at actual_delivery_time  store_id  \
0             1.0  2015-02-06 22:24:17  2015-02-06 23:27:16      1845   
1             2.0  2015-02-10 21:49:25  2015-02-10 22:56:29      5477   
8             2.0  2015-02-16 00:11:35  2015-02-16 00:38:01      5477   
14            1.0  2015-02-12 03:36:46  2015-02-12 04:14:39      2841   
15            1.0  2015-01-27 02:12:36  2015-01-27 03:02:24      2841   
...           ...                  ...                  ...       ...   
197423        1.0  2015-02-17 00:19:41  2015-02-17 01:24:48      2956   
197424        1.0  2015-02-13 00:01:59  2015-02-13 00:58:22      2956   
197425        1.0  2015-01-24 04:46:08  2015-01-24 05:36:16      2956   
197426        1.0  2015-02-01 18:18:15  2015-02-01 19:23:22      3630   
197427        1.0  2015-02-08 19:24:33  2015-02-08 20:01:41      3630   

       store_primary_category  order_protocol  total_items  subtotal  \
0                    american             1.0            4      3441   
1                     mexican             2.0            1      1900   
8                      indian             3.0            4      4771   
14                    italian             1.0            1      1525   
15                    italian             1.0            2      3620   
...                       ...             ...          ...       ...   
197423                   fast             4.0            3      1389   
197424                   fast             4.0            6      3010   
197425                   fast             4.0            5      1836   
197426               sandwich             1.0            1      1175   
197427               sandwich             1.0            4      2605   

        num_distinct_items  min_item_price  max_item_price  \
0                        4             557            1239   
1                        1            1400            1400   
8                        3             820            1604   
14                       1            1525            1525   
15                       2            1425            2195   
...                    ...             ...             ...   
197423                   3             345             649   
197424                   4             405             825   
197425                   3             300             399   
197426                   1             535             535   
197427                   4             425             750   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
0                        33.0                14.0                      21.0   
1                         1.0                 2.0                       2.0   
8                         8.0                 6.0                      18.0   
14                        5.0                 6.0                       8.0   
15                        5.0                 5.0                       7.0   
...                       ...                 ...                       ...   
197423                   17.0                17.0                      23.0   
197424                   12.0                11.0                      14.0   
197425                   39.0                41.0                      40.0   
197426                    7.0                 7.0                      12.0   
197427                   20.0                20.0                      23.0   

        estimated_order_place_duration  \
0                                  446   
1                                  446   
8                                  446   
14                                 446   
15                                 446   
...                                ...   
197423                             251   
197424                             251   
197425                             251   
197426                             446   
197427                             446   

        estimated_store_to_consumer_driving_duration      sum  time

In [36]:
datos13=datos12[(datos12['times']<22) & (datos12['times']>16) & (datos12['market_id']==6)]

filtramos los datos de la region con menos pedidos entre las horas de 17h-22h siendo (5pm a 10pm)

In [37]:
datos13

market_id           created_at actual_delivery_time  store_id  \
525           6.0  2015-02-04 19:49:41  2015-02-04 20:29:05      2716   
1474          6.0  2015-02-13 18:24:10  2015-02-13 18:51:39       517   
2298          6.0  2015-01-25 19:15:34  2015-01-25 20:01:30      5711   
4909          6.0  2015-02-16 21:55:14  2015-02-16 22:48:52      4221   
6075          6.0  2015-01-30 19:24:50  2015-01-30 20:48:58      2722   
...           ...                  ...                  ...       ...   
194463        6.0  2015-02-14 20:32:10  2015-02-14 21:04:17      1409   
194944        6.0  2015-02-04 19:42:32  2015-02-04 20:33:28      4756   
195694        6.0  2015-02-02 18:33:25  2015-02-02 19:28:50      5201   
195829        6.0  2015-01-28 19:21:29  2015-01-28 20:08:24      4862   
197067        6.0  2015-01-23 20:37:37  2015-01-23 21:15:08      1409   

       store_primary_category  order_protocol  total_items  subtotal  \
525                      cafe             3.0            1      1008   
1474                  seafood             3.0            2      2220   
2298                     cafe             1.0            1      1478   
4909                  italian             3.0            2      3000   
6075                  seafood             4.0            4      2648   
...                       ...             ...          ...       ...   
194463                   thai             5.0            2      1060   
194944                mexican             1.0            4      2498   
195694                   thai             5.0            2      1150   
195829                mexican             3.0            2      1897   
197067               american             1.0            2      1070   

        num_distinct_items  min_item_price  max_item_price  \
525                      1             849             849   
1474                     2            1095            1125   
2298                     1            1478            1478   
4909                     2            1400            1600   
6075                     2               0             875   
...                    ...             ...             ...   
194463                   2             165             895   
194944                   3             199             799   
195694                   2             250             900   
195829                   2             898             999   
197067                   2             195             750   

        total_onshift_dashers  total_busy_dashers  total_outstanding_orders  \
525                      39.0                39.0                      59.0   
1474                      9.0                 8.0                       8.0   
2298                     16.0                16.0                      21.0   
4909                     15.0                14.0                      11.0   
6075                      8.0                 9.0                      13.0   
...                       ...                 ...                       ...   
194463                   49.0                48.0                      64.0   
194944                   40.0                39.0                      57.0   
195694                    1.0                 1.0                       2.0   
195829                   31.0                31.0                      53.0   
197067                   71.0                57.0                      65.0   

        estimated_order_place_duration  \
525                                251   
1474                               251   
2298                               251   
4909                               446   
6075                               251   
...                                ...   
194463                             251   
194944                             251   
195694                             251   
195829                             251   
197067                             251   

        estimated_store_to_consumer_driving_duration      sum  time

In [38]:
datos14=datos13.groupby('times')

hacemos una agrupacion de los tiempos

In [39]:
datos13['count']=datos14['times'].transform('count')

<ipython-input-39-026ba1722e1a>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  datos13['count']=datos14['times'].transform('count')


contamos la cantidad de los tiempos de los pedidos

In [40]:
datos15=datos13[['times','count']]

In [41]:
datos16=datos15.set_index('count', inplace=True)

separamos los times y contadores

In [42]:
datos15

times
count       
43        19
12        18
43        19
33        21
43        19
...      ...
41        20
43        19
12        18
43        19
41        20

[144 rows x 1 columns]

In [43]:
datos15=datos15.drop_duplicates()
datos15

times
count       
43        19
12        18
33        21
41        20
15        17

ponemos como indices los count y eliminamos los datos duplicados asi obtenemos de 17-22 la hora  donde se realizan menor cantidad de pedidos, y seria la mejor hora para ordenar siendo 18h con la menor cantidad de pedidos